packages/https-proxy/test/unit/ca_spec.js

```diff
@@ -1,161 +0,0 @@
-/* eslint-disable no-console */
-const { expect } = require('../spec_helper')
-
-let fs = require('fs-extra')
-const path = require('path')
-const Promise = require('bluebird')
-const CA = require('../../lib/ca')
-
-fs = Promise.promisifyAll(fs)
-
-describe('lib/ca', () => {
-  beforeEach(function () {
-    this.timeout(5000)
-
-    this.dir = path.join(process.cwd(), 'tmp')
-
-    return fs.ensureDirAsync(this.dir)
-    .then(() => {
-      console.time('creating CA')
-
-      return CA.create(this.dir)
-      .tap(() => {
-        console.timeEnd('creating CA')
-      })
-    }).then((ca) => {
-      this.ca = ca
-    })
-  })
-
-  afterEach(function () {
-    return fs.removeAsync(this.dir)
-  })
-
-  context('#generateServerCertificateKeys', () => {
-    it('generates certs for each host', async function () {
-      console.time('generating cert')
-
-      await this.ca.generateServerCertificateKeys('www.cypress.io')
-      .spread((certPem, keyPrivatePem) => {
-        console.timeEnd('generating cert')
-        expect(certPem).to.include('-----BEGIN CERTIFICATE-----')
-
-        expect(keyPrivatePem).to.include('-----BEGIN RSA PRIVATE KEY-----')
-      })
-    })
-  })
-
-  context('.create', () => {
-    it('returns a new CA instance', function () {
-      expect(this.ca).to.be.an.instanceof(CA)
-    })
-
-    it('creates certs + keys dir', function () {
-      return Promise.join(
-        fs.statAsync(path.join(this.dir, 'certs')),
-        fs.statAsync(path.join(this.dir, 'keys')),
-      )
-    })
-
-    it('writes certs/ca.pem', function () {
-      return fs.statAsync(path.join(this.dir, 'certs', 'ca.pem'))
```

Comment:
most of these tests weren't doing anything useful, so the tests themselves were overhauled to test the main `.create` scenarios